### DISCUSSION TITLE
Stepper Index

### DISCUSSION THREAD

**Discussion by claudiaareneee** [iTwin/iTwinUI#1692] [Q&A]
Hey! Does anyone know how to change the first number/text of the stepper component?

I have this stepper:
![image](https://github.com/iTwin/iTwinUI/assets/25939215/ba945790-bdbf-4632-80ad-9720906d4a4d)

I want the `1` to read `Intro` , `2` to read `1`, etc. ...

Any advice on how to do that?
Thanks! 
Claudia

**Comment by mayank99** [comment]
I don't think this is currently possible. You'll have to show the name underneath the number.

```tsx
<Stepper
  currentStep={1}
  steps={[
    { name: 'Intro' },
    { name: 'Current' },
    { name: 'Next' },
  ]}
/>
```

Is there a reason you want to change the design?

**Reply by claudiaareneee** [reply]
Yeah so this is what I've got. We're using the stepper to show progress through a coding challenge: 
![image](https://github.com/iTwin/iTwinUI/assets/25939215/bde7cf79-f7b3-4400-a45a-2f59a1dff558)
The issue is the number miss match between what's in the circle and what's underneath. This is confusing to users

**Reply by mayank99** [reply]
would something like this work? zero-based indexing
![](https://github.com/iTwin/iTwinUI/assets/9084735/e4244f7e-745c-4d7a-9b1b-054294d52fa7)

we can support that in v3 if you're willing to upgrade

**Reply by claudiaareneee** [reply]
That would work for me. Or even just a prop to override what's in the circle. (I really want to upgrade, but it's going to take effort. We're really behind. Looking for more reasons to upgrade :) )

**Reply by mayank99** [reply]
v1/v2 will not get any new features, but we can totally support this in v3 (in the near future!)

Let us know when you're ready to upgrade and if you get stuck/need any guidance.
